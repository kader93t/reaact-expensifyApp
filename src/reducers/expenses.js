// Expenses Reducer

const expenseDefaultState = [];
export default (state = expenseDefaultState, action) => {
    
    switch (action.type) {
        case "ADD_EXPENSE":
            return [       
                ...state, action.expenses
            ]
        case "REMOVE_EXPENSE":
            return state.filter((expense) => {
                return expense.id !== action.id;
            })
        case "UPDATE_EXPENSE":
            return state.map((expense) => {
                if (expense.id === action.id) {
                    return {
                        ...expense,
                        ...action.updates
                    }
                } else {
                    return expense;
                }
            });
            ;
        default:
            return state;
    }
}


