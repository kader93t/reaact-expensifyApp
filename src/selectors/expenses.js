
export default (expenses, { startDate, endDate, text, sortBy }) => {
    const visibleExpenses = expenses.filter((expense) => {
        const matchStartDate = !startDate || startDate.isSameOrBefore(expense.createAt, "day");
        const matchEndDate = !endDate || endDate.isSameOrAfter(expense.createAt, "day");
        const matchText = !text || expense.description.toLowerCase().includes(text.toLowerCase());
        return matchEndDate && matchStartDate && matchText;
    });
    return visibleExpenses.sort((a, b) => {
        if (sortBy === "date")
            return a.createAt < b.createAt ? 1 : -1;
        else if (sortBy === "amount")
            return a.amount < b.amount ? 1 : -1;
    });
};
