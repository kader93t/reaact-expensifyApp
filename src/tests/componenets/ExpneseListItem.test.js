import React from 'react'
import ExpenseListItem from "./../../components/ExpenseItem";
import { shallow } from "enzyme";
import {expenses} from "./../fixture/expenses"

test("should render the item ", () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
});
