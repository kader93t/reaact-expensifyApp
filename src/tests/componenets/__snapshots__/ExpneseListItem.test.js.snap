// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the item  1`] = `
<Link
  className="list-item"
  to="/edite/1"
>
  <div
    className=""
  >
    <h3
      className="list-item__title"
    >
      hello1
    </h3>
    <span
      className="list-item__sub-title"
    >
      1st January 1970
    </span>
  </div>
  <div>
    <h3
      className="list-item__amount"
    >
      $2.12
    </h3>
  </div>
</Link>
`;
